app/configurator/configurator-state.tsx

```diff
@@ -636,15 +639,32 @@ export const canTransitionToPreviousStep = (_: ConfiguratorState): boolean => {
   return true;
 };
 
+// FIXME: should by handled better, as color is a subfield and not actual field.
+// Side effects in ui encodings?
+export const extractCategoricalColorIris = (
+  chartConfig: ChartConfig
+): string[] => {
+  const iris: string[] = [];
+
+  if (isColorFieldInConfig(chartConfig)) {
+    if (chartConfig.fields.areaLayer?.color.type === "categorical") {
+      iris.push(chartConfig.fields.areaLayer.color.componentIri);
+    }
+
+    if (chartConfig.fields.symbolLayer?.color.type === "categorical") {
+      iris.push(chartConfig.fields.symbolLayer.color.componentIri);
+    }
+  }
+
+  return iris;
+};
+
 export const getFiltersByMappingStatus = (chartConfig: ChartConfig) => {
   const genericFieldValues = Object.values(chartConfig.fields).map(
     (d) => d.componentIri
   );
-  const nonGenericFieldValues =
-    isMapConfig(chartConfig) &&
-    chartConfig.fields.symbolLayer?.color.type === "categorical"
-      ? [chartConfig.fields.symbolLayer.color.componentIri]
-      : [];
+  const nonGenericFieldValues = extractCategoricalColorIris(chartConfig);
```

Comment:
Nit: I would call the function extractNonGenericFields, I think it could handle other cases than only the categorical color. Here my brain does "why the left is called nonGenericField and the right is called categorical colors".

Reply:
Good point 👍